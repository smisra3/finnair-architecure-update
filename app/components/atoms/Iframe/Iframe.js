// @flow
import { PureComponent } from 'react';
import styles from './Iframe.style';
import styled from 'styled-components';

type Props = {
  className: string,
  sourceUrl: string,
  iframeTitle: string,
  inheritedStyles?: string | {},
  onLoad?: (SyntheticEvent<>) => void,
  width?: string,
  height?: string,
  takeContentHeight?: boolean,
  frameBorder?: string,
};

type State = {
  frameHeight: string,
};

class Iframe extends PureComponent<Props, State> {
  static defaultProps = {
    inheritedStyles: '',
    width: '100%',
    height: '100%',
    frameBorder: '0',
    takeContentHeight: false,
    onLoad: () => {},
  };

  state = {
    frameHeight: '100%',
  };

  iframe = <iframe title="no iframe" />;

  render() {
    const {
      className,
      sourceUrl,
      iframeTitle,
      inheritedStyles,
      onLoad,
      width,
      height,
      takeContentHeight,
      frameBorder,
      ...others
    }: Props = this.props;
    return (
      <iframe
        ref={(iframe) => {
          this.iframe = iframe;
        }}
        className={className}
        title={iframeTitle}
        src={sourceUrl}
        width={width}
        height={this.props.takeContentHeight ? this.state.frameHeight : height}
        frameBorder={frameBorder}
        onLoad={(event) => {
          if (takeContentHeight) {
            this.setState({
              frameHeight: `${this.iframe.contentWindow.document.body.scrollHeight}px`,
            });
          }
          if (typeof onLoad === 'function') {
            onLoad(event);
          }
        }}
        {...others}
      />
    );
  }
}

export default styled(Iframe)`${styles}`;
export { Iframe as IframeComponent };
