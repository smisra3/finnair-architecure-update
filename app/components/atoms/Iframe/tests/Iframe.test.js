import { shallow } from 'enzyme';
import { IframeComponent } from '../Iframe';

describe('Iframe', () => {
  let iframe = '';

  test('should render correctly', () => {
    expect(iframe).toMatchSnapshot();
  });

  test('should render additional props', () => {
    iframe = shallow(<IframeComponent
      iframeTitle="Hello iframe"
      sourceUrl="http://www.google.com"
      name="testIframe"
    />);
    expect(iframe.find({ name: 'testIframe' })).toHaveLength(1);
  });

  test('should take ContentHeight', () => {
    iframe = shallow(<IframeComponent
      iframeTitle="Hello iframe"
      sourceUrl="http://www.google.com"
      name="testIframe"
      takeContentHeight="234px"
    />);
    expect(iframe).toMatchSnapshot();
  });

  test('should call onLoad Method', () => {
    iframe = shallow(<IframeComponent
      iframeTitle="Hello iframe"
      sourceUrl="http://www.google.com"
      name="testIframe"
      takeContentHeight="234px"
      onLoad={() => {}}
    />);
    iframe.instance().iframe = { contentWindow: { document: { body: { scrollHeight: 20 } } } };
    iframe.find('iframe').simulate('load');
    expect(iframe).toMatchSnapshot();
  });
});
