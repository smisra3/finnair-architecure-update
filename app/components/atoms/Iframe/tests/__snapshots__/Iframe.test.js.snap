// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Iframe should call onLoad Method 1`] = `
<iframe
  frameBorder="0"
  height="20px"
  name="testIframe"
  onLoad={[Function]}
  src="http://www.google.com"
  title="Hello iframe"
  width="100%"
/>
`;

exports[`Iframe should render correctly 1`] = `""`;

exports[`Iframe should take ContentHeight 1`] = `
<iframe
  frameBorder="0"
  height="100%"
  name="testIframe"
  onLoad={[Function]}
  src="http://www.google.com"
  title="Hello iframe"
  width="100%"
/>
`;
